src/label/Label.ts

```diff
@@ -0,0 +1,77 @@
+import { WidgetBase } from '@dojo/widget-core/WidgetBase';
+import { ThemeableMixin, ThemeableProperties, theme } from '@dojo/widget-core/mixins/Themeable';
+import { v } from '@dojo/widget-core/d';
+import { assign } from '@dojo/core/lang';
+import * as css from './styles/label.css';
+
+/**
+ * Label settings for form label text content, position (before or after), and visibility
+ */
+export interface LabelOptions {
+	content: string;
+	before?: boolean;
+	hidden?: boolean;
+}
+
+/**
+ * Default settings for form labels
+ */
+const labelDefaults = {
+	content: '',
+	before: true,
+	hidden: false
+};
+
+/**
+ * @type LabelWidgetProperties
+ *
+ * Properties that can be set on a Label component
+ *
+ * @property classes	Optional classes to be set on the label node
+ * @property formId		ID of a form element associated with the form field
+ * @property label		Label settings for form label text, position, and visibility
+ */
+export interface LabelProperties extends ThemeableProperties {
+	classes?: { [key: string]: boolean };
```

Comment:
I was expecting users to pass `this.classes(...)` in to `label.properties.classes`, yes. Is there a different or stricter way you'd prefer?